modules/swagger-codegen/src/main/resources/go/api.mustache

```diff
@@ -43,6 +43,28 @@ func (a {{classname}}) {{nickname}} ({{#allParams}}{{paramName}} {{{dataType}}}{
 
     _sling := sling.New().{{httpMethod}}(a.Configuration.BasePath)
 
+    {{#authMethods}}// authentication ({{name}}) required
+    {{#isApiKey}}{{#isKeyInHeader}}
+    // set key with prefix in header
+    _sling.Set("{{keyParamName}}", a.Configuration.GetApiKeyWithPrefix("{{keyParamName}}")
+    {{/isKeyInHeader}}{{#isKeyInQuery}}
+    // set key with prefix in querystring
+    _sling.Set("{{keyParamName}}",  a.Configuration.GetApiKeyWithPrefix("{{keyParamName}}")
```

Comment:
Yes, you need to use the QueryParams struct: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/go/api.mustache#L58-L63